tools/c7n_azure/c7n_azure/handler.py

```diff
@@ -31,6 +28,10 @@
 
 
 def run(event, context, subscription_id=None):
+    # Always reset session cache before the run.. If there is any issue with configuration,
+    # if next run happens in the same process, it might have different target subscription id.
```

Comment:
(nit) "Custodian can be executed in the same process, but target different subscriptions. Consequently, we need to reset the session."